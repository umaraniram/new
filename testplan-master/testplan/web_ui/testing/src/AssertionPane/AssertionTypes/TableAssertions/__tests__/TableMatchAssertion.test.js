import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import TableMatchAssertion from '../TableMatchAssertion';

function defaultProps() {
  return {
    assertion: {
      "category":"DEFAULT",
      "machine_time":"2019-02-12T17:41:43.176922+00:00",
      "description":"Table Match: list of list vs list of list",
      "exclude_columns":null,
      "report_fails_only":false,
      "fail_limit":0,
      "line_no":284,
      "data":[
        [0,["Bob",32],{},{},{}],
        [1,["Susan",24],{},{},{}],
        [2,["Rick",67],{},{},{}]
      ],
      "strict":false,
      "meta_type":"assertion",
      "columns":["name","age"],
      "passed":true,
      "include_columns":null,
      "message":null,
      "type":"TableMatch",
      "utc_time":"2019-02-12T17:41:43.176916+00:00"
    }
  };
}


describe('FixMatchAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<TableMatchAssertion {...props}/>);
    expect(shallowComponent).toMatchSnapshot();
  });
});