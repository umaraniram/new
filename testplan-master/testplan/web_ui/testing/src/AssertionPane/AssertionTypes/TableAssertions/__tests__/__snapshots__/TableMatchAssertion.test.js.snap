// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixMatchAssertion shallow renders the correct HTML structure 1`] = `
<TableBaseAssertion
  columns={
    Array [
      Object {
        "cellStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
          "width": "72px",
        },
        "field": "id",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "width": "72px",
        },
        "title": "ID",
      },
      Object {
        "cellStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "fontSize": "small",
          "padding": "6px 11px 6px 11px",
          "width": "121px",
        },
        "field": "ev",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
          "width": "121px",
        },
        "title": "Expected/Value",
      },
      Object {
        "cellStyle": [Function],
        "field": "name",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "title": "name",
      },
      Object {
        "cellStyle": [Function],
        "field": "age",
        "headerStyle": Object {
          "border": "1px solid",
          "borderColor": "#d9dcde",
        },
        "title": "age",
      },
    ]
  }
  rows={
    Array [
      Object {
        "age": 32,
        "ev": "Expected",
        "id": 0,
        "name": "Bob",
        "passed": Object {
          "age": true,
          "name": true,
        },
      },
      Object {
        "age": 32,
        "ev": "Value",
        "id": 0,
        "name": "Bob",
        "passed": Object {
          "age": true,
          "name": true,
        },
      },
      Object {
        "age": 24,
        "ev": "Expected",
        "id": 1,
        "name": "Susan",
        "passed": Object {
          "age": true,
          "name": true,
        },
      },
      Object {
        "age": 24,
        "ev": "Value",
        "id": 1,
        "name": "Susan",
        "passed": Object {
          "age": true,
          "name": true,
        },
      },
      Object {
        "age": 67,
        "ev": "Expected",
        "id": 2,
        "name": "Rick",
        "passed": Object {
          "age": true,
          "name": true,
        },
      },
      Object {
        "age": 67,
        "ev": "Value",
        "id": 2,
        "name": "Rick",
        "passed": Object {
          "age": true,
          "name": true,
        },
      },
    ]
  }
/>
`;
